notification-hubs-sdk/src/fcm/java/com/microsoft/windowsazure/messaging/NotificationHubLegacyExtension.java

```diff
@@ -0,0 +1,66 @@
+package com.microsoft.windowsazure.messaging;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.AsyncTask;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.messaging.FirebaseMessaging;
+
+public class NotificationHubLegacyExtension {
+    private static final String MIGRATED_TO_FCM_V1_KEY = "__ANH__MIGRATED_TO_FCM_V1";
+
+    public static Task<Void> migrateToFcmV1(Context context, NotificationHub hub, String ...tags) {
```

Comment:
I love it - are we going to deprecate this after July when Google has truly turned the servers off of the legacy FCM APIs?

Reply:
Yes, that is correct.

Reply:
Note that this is only needed for customers using the Registration flow